src/main/java/com/bookshop/controller/BookController.java

```diff
@@ -23,14 +24,15 @@
 import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestController;
 
-@Tag(name = "Book store API", description = "Endpoints for managing books")
 @RequiredArgsConstructor
 @RestController
 @Validated
-@RequestMapping(value = "/books")
+@RequestMapping(value = "/api/books")
+@Tag(name = "Book store API", description = "Endpoints for managing books")
 public class BookController {
     private final BookService bookService;
 
+    @PreAuthorize("hasAnyRole('ROLE_ADMIN', 'ROLE_USER')")
```

Comment:
1. `hasRole` adds `ROLE_` prefix automatically - you may not specify it
2. doesn't ADMIN has all the priviliges USER has? if yes - then you can simplify to `hasRole("USER")`. if not - feel free to skip it